src/applications/static-pages/sharable-links/sharableLink.js

```diff
@@ -0,0 +1,170 @@
+  color: ${props => props.theme.colorWhite};
+  font-family: 'Source Sans Pro';
+  font-weight: 400;
+  font-size: 12px;
+  height: 26px;
+  padding: 4px;
+  border: 1px solid;
+  bottom: ${props => (props.leftAligned ? '' : '1px')};
+`;
+
+const ShareIcon = styled.i`
+  border-radius: 5px;
+  font-size: 16px;
+  height: 26px;
+  width: 26px;
+  padding: 4px;
+  border: 1px solid;
+  background-color: ${props =>
+    props.feedbackActive ? props.theme.colorBaseBlack : ''};
+  color: ${props =>
+    props.feedbackActive ? props.theme.colorWhite : props.theme.colorPrimary};
+
+  &:hover {
+    background-color: ${props => props.theme.colorBaseBlack};
+    color: ${props => props.theme.colorWhite};
+    cursor: pointer;
+  }
+`;
+
+// need this to override this global css rule coming from somewhere:
+// .usa-accordion > ul button, .usa-accordion-bordered > ul button
+
+const UnStyledButtonInAccordion = styled.button`
+  padding: 0px !important;
+  background-color: transparent !important;
+  background-image: none !important;
+  width: auto !important;
+`;
+
+const SharableLink = ({ dataEntityId }) => {
+  const [feedbackActive, setFeedbackActive] = useState(false);
+  const [copiedText] = useState('Link copied');
+  const [leftAligned, setLeftAligned] = useState(false);
+  const [leftPx, setLeftPx] = useState(0);
+  const offsetThreshold = 100;
+  const widthOffset = 40;
+
+  const extractId = idString => {
+    if (!idString) return '';
+    const arr = idString.split('-');
+    arr.shift();
+    return arr.join('-');
+  };
+
+  const hidePreviousFeedbacks = activeId => {
+    const otherActiveFeedbacks = document.getElementsByClassName(
+      'sharable-link-feedback',
+    );
+
+    for (const feedback of otherActiveFeedbacks) {
+      const parentId = feedback.getAttribute('id');
+      if (extractId(parentId) !== extractId(activeId)) {
+        feedback.style.display = 'none';
+      } else {
+        feedback.style = {};
+      }
+    }
+  };
+
+  const hideFeedback = activeId => {
+    hidePreviousFeedbacks(activeId);
+    setTimeout(() => {
+      setFeedbackActive(false);
+      setLeftAligned(false);
+      setLeftPx(0);
+    }, 10000);
+  };
+
+  const displayFeedback = element => {
+    const iconParentId = extractId(element.getAttribute('id'));
+    const parentElement = document.getElementById(iconParentId);
+
+    if (
+      parentElement?.offsetWidth - (element.offsetLeft + element.offsetWidth) <=
+      offsetThreshold
+    ) {
+      setLeftAligned(true);
+      setLeftPx(element.offsetLeft - element.offsetWidth - widthOffset);
+    }
+    setFeedbackActive(true);
+    hideFeedback(element.getAttribute('id'));
+  };
+
+  return (
+    <ThemeProvider theme={theme.main}>
+      <span aria-live="polite" aria-relevant="additions">
+        <UnStyledButtonInAccordion
+          className="usa-button-unstyled"
+          aria-label={`Copy ${dataEntityId} sharable link`}
+        >
+          <ShareIcon
```

Comment:
This would need to be a `<a>` or `<button>` (focusable element) so the click action can be accessible to keyboard users

Reply:
it's wrapped in a button

Reply:
i thought aria-hidden="true" was needed for icons